internal/ad/registry/registry_test.go

```diff
@@ -427,9 +426,25 @@ func FuzzDecodePolicy(f *testing.F) {
 		f.Add(d)
 	}
 
-	f.Fuzz(func(_ *testing.T, d []byte) {
-		r := bytes.NewReader(d)
-		_, _ = registry.DecodePolicy(r)
+	// Setup a temporary directory to store the fuzzed files
+	fuzzDir, err := os.MkdirTemp("", f.Name())
+	if err != nil {
+		f.Fail()
+	}
+	f.Cleanup(func() { os.RemoveAll(fuzzDir) })
+
+	f.Fuzz(func(t *testing.T, d []byte) {
```

Comment:
After running this fuzz test for 1m20s, it found an input which resulted in an error:

```
fuzz: elapsed: 1m20s, execs: 1429611 (0/sec), new interesting: 5 (total: 81)
--- FAIL: FuzzDecodePolicy (79.89s)
    fuzzing process hung or terminated unexpectedly while minimizing: EOF
    Failing input written to testdata/fuzz/FuzzDecodePolicy/44ef800113434b04
    To re-run:
    go test -run=FuzzDecodePolicy/44ef800113434b04
```

[Here](https://github.com/user-attachments/files/19434212/44ef800113434b04.zip) is the failing input which you can use to reproduce and debug the failure. To reproduce, put the input in `internal/ad/registry/testdata/fuzz/FuzzDecodePolicy` and run `go test -fuzz FuzzDecodePolicy ./internal/ad/registry/...`.